ext/ddtrace_profiling_native_extension/collectors_cpu_and_wall_time_worker.c

```diff
@@ -846,22 +877,43 @@ static VALUE _native_stats(DDTRACE_UNUSED VALUE self, VALUE instance) {
-    ID2SYM(rb_intern("skipped_sample_because_of_dynamic_sampling_rate")), /* => */ UINT2NUM(state->stats.skipped_sample_because_of_dynamic_sampling_rate),
     ID2SYM(rb_intern("postponed_job_skipped_already_existed")),      /* => */ UINT2NUM(state->stats.postponed_job_skipped_already_existed),
     ID2SYM(rb_intern("postponed_job_success")),                      /* => */ UINT2NUM(state->stats.postponed_job_success),
     ID2SYM(rb_intern("postponed_job_full")),                         /* => */ UINT2NUM(state->stats.postponed_job_full),
     ID2SYM(rb_intern("postponed_job_unknown_result")),               /* => */ UINT2NUM(state->stats.postponed_job_unknown_result),
-    ID2SYM(rb_intern("sampling_time_ns_min")),                       /* => */ pretty_sampling_time_ns_min,
-    ID2SYM(rb_intern("sampling_time_ns_max")),                       /* => */ pretty_sampling_time_ns_max,
-    ID2SYM(rb_intern("sampling_time_ns_total")),                     /* => */ pretty_sampling_time_ns_total,
-    ID2SYM(rb_intern("sampling_time_ns_avg")),                       /* => */ pretty_sampling_time_ns_avg,
+    ID2SYM(rb_intern("cpu_sampling_time_ns_min")),                   /* => */ pretty_cpu_sampling_time_ns_min,
+    ID2SYM(rb_intern("cpu_sampling_time_ns_max")),                   /* => */ pretty_cpu_sampling_time_ns_max,
+    ID2SYM(rb_intern("cpu_sampling_time_ns_total")),                 /* => */ pretty_cpu_sampling_time_ns_total,
+    ID2SYM(rb_intern("cpu_sampling_time_ns_avg")),                   /* => */ pretty_cpu_sampling_time_ns_avg,
+    ID2SYM(rb_intern("cpu_sleeping_time_ns_avg")),                   /* => */ pretty_cpu_sleeping_time_ns_avg,
+    ID2SYM(rb_intern("allocation_sampling_time_ns_min")),            /* => */ pretty_allocation_sampling_time_ns_min,
+    ID2SYM(rb_intern("allocation_sampling_time_ns_max")),            /* => */ pretty_allocation_sampling_time_ns_max,
+    ID2SYM(rb_intern("allocation_sampling_time_ns_total")),          /* => */ pretty_allocation_sampling_time_ns_total,
+    ID2SYM(rb_intern("allocation_sampling_time_ns_avg")),            /* => */ pretty_allocation_sampling_time_ns_avg,
     ID2SYM(rb_intern("allocations_during_sample")),                  /* => */ UINT2NUM(state->stats.allocations_during_sample),
+    ID2SYM(rb_intern("allocation_sampler_event_rate")),              /* => */ DBL2NUM(discrete_dynamic_sampler_event_rate(state->allocation_sampler)),
+    ID2SYM(rb_intern("allocation_sampler_probability")),             /* => */ DBL2NUM(discrete_dynamic_sampler_probability(state->allocation_sampler)),
+    ID2SYM(rb_intern("allocation_sampler_time_ns")),                 /* => */ ULONG2NUM(discrete_dynamic_sampler_sampling_time_ns(state->allocation_sampler)),
+    ID2SYM(rb_intern("allocation_sampler_target_overhead_adj")),     /* => */ DBL2NUM(discrete_dynamic_sampler_target_overhead_adjustment(state->allocation_sampler)),
```

Comment:
Yeah I was actually fighting between the two at the beginning and went with this one to try to keep the whole sampler Ruby "pure". But it would simplify a bunch of things indeed so I'll play with the alternative.

Reply:
https://datadoghq.atlassian.net/browse/PROF-9036